### ISSUE TITLE
Error type does not show as error

### ISSUE THREAD

**Issue by yigit** [google/ksp#371]
Hit this issue while converting a room test to also run with KSP.
https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:room/compiler/src/test/kotlin/androidx/room/processor/PojoProcessorTest.kt;l=477?q=PojoProcessorTest

```
package foo.bar;
import androidx.room.*;
import java.util.*;
public class MyPojo {
  int id;
  @Relation(parentColumn = "id", entityColumn = "uid")
  public List<User> user;
}
```

When I resolve the type of `user` via `asMemberOf` (`MyPojo`) and then get the type argument type (from List), its `isError` check returns `false` instead of `true`.

When I debug, the `ksType` is `(User..User?)`  and its `declaration.descriptor` is instance of `NotFoundClasses$MockClassDescriptor` but somehow KSP is missing it.

We probably need to expand this check:
https://github.com/google/ksp/blob/master/compiler-plugin/src/main/kotlin/com/google/devtools/ksp/symbol/impl/java/KSTypeReferenceJavaImpl.kt#L98

to always check the declaration?

I'll try to followup with a repro in KSP test infra.

**Comment by yigit** [comment]
Also, this looks like very similar to:
https://github.com/google/ksp/issues/107 , might even be the same issue.

Appearantly, we did work around it in room.
https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:room/compiler-processing/src/main/java/androidx/room/compiler/processing/ksp/KSAsMemberOf.kt;l=35?q=KSAsMemberOf

I might be able to expand that to also check type arguments but we should eventually fix this in KSP.

**Comment by yigit** [comment]
oh actually, in this particular codepath, we may not be running through as Member of, i need to check in more detail.

**Comment by neetopia** [comment]
can you give me more detail on this? I tried to call `asMemberOf` on a `typePair: Pair<BaseTypeArg2, NonExist>` and it is returning `isError=true` for the second type argument in the return result of `asMemberOf`. Is my understanding of the issue here correctly?

**Comment by neetopia** [comment]
fixed in #607
